src/main/java/org/opensearch/ad/transport/handler/AnomalyResultBulkIndexHandler.java

```diff
@@ -123,47 +115,30 @@ private void bulkSaveDetectorResult(List<AnomalyResult> anomalyResults, ActionLi
     }
 
     private void bulkSaveDetectorResult(String resultIndex, List<AnomalyResult> anomalyResults, ActionListener<BulkResponse> listener) {
-        List<BulkOperation> operations = new ArrayList<>();
+        BulkRequest bulkRequest = new BulkRequest();
         anomalyResults.forEach(anomalyResult -> {
-            BulkOperation operation = new BulkOperation.Builder().index(i -> i.index(resultIndex).document(anomalyResult)).build();
-            operations.add(operation);
+            try (XContentBuilder builder = jsonBuilder()) {
+                IndexRequest indexRequest = new IndexRequest(resultIndex)
+                    .source(anomalyResult.toXContent(builder, RestHandlerUtils.XCONTENT_WITH_TYPE));
+                bulkRequest.add(indexRequest);
+            } catch (Exception e) {
+                String error = "Failed to prepare request to bulk index anomaly results";
+                LOG.error(error, e);
+                throw new AnomalyDetectionException(error);
+            }
         });
-        BulkRequest bulkRequest = new BulkRequest.Builder().operations(operations).build();
-        try {
-            CompletableFuture<BulkResponse> response = sdkJavaAsyncClient.bulk(bulkRequest);
-            BulkResponse bulkResponse = response
-                .orTimeout(AnomalyDetectorSettings.REQUEST_TIMEOUT.get(settings).getMillis(), TimeUnit.MILLISECONDS)
-                .get();
-
-            if (bulkResponse.errors()) {
-                StringBuilder sb = new StringBuilder();
-                sb.append("Failed to prepare request to bulk index anomaly results:");
-                for (int i = 0; i < bulkResponse.items().size(); i++) {
-                    BulkResponseItem item = bulkResponse.items().get(i);
-                    if (item.error() != null) {
-                        sb
-                            .append("\n[")
-                            .append(i)
-                            .append("]: index [")
-                            .append(item.index())
-                            .append("], id [")
-                            .append(item.id())
-                            .append("], message [")
-                            .append(item.error().reason())
-                            .append("]");
-                    }
-                }
-                String failureMessage = sb.toString();
+        client.bulk(bulkRequest, ActionListener.wrap(r -> {
+            if (r.hasFailures()) {
+                String failureMessage = r.buildFailureMessage();
```

Comment:
Since you have already translated the request to java client. Why changing it back here?

Reply:
I made the original modification before implementing the historical analysis workflow. Now that we have enabled this workflow, during testing I faced an error with bulk requests using the java client with an index initialized using the AD `clientUtil`, which internally uses the `SDKRestClient` to send a create index request. To circumnavigate this issue, I have opted to revert my previous changes and use the `bulk` method of the `SDKRestClient` to perform this request.